src/Datadog.Trace/Configuration/GlobalSettings.cs

```diff
@@ -121,15 +131,28 @@ internal static CompositeConfigurationSource CreateDefaultConfigurationSource()
             return configurationSource;
         }
 
-        private static bool TryLoadJsonConfigurationFile(IConfigurationSource configurationSource, out IConfigurationSource jsonConfigurationSource)
+        private static bool TryLoadPluginJsonConfigurationFile(IConfigurationSource configurationSource, out JsonConfigurationSource jsonConfigurationSource)
+        {
+            var configurationFileName = configurationSource?.GetString(ConfigurationKeys.PluginConfigurationFileName) ??
+                                        Path.Combine(GetCurrentDirectory(), "plugins.json");
```

Comment:
Not sure if we want to use `Path.Combine(GetCurrentDirectory(), "plugins.json")` if nothing is provided.
I always fear such stuff because it is a great place that someone may use for ACE (Arbitrary Code Execution). The bad-guy must only have a possibility to create some file in the same directory (it can be created by a different user!). The user may be not aware of the risk. I would prefer to require explicitly enabling via ENV VAR. Non-admin user is unable to set env var for different users.

Reply:
Initially almost the same logic as for json configuration file, just tries to load different file (see `TryLoadJsonConfigurationFile`). Just removing `plugins.json` defaults wouldn't give much benefits, as the ACE could be executed through autoloading `datadog.json` which in turn can direct to plugin by overloading unspecified plugin path.

```
var configurationFileName = configurationSource.GetString(ConfigurationKeys.ConfigurationFileName) ??
      configurationSource.GetString("OTEL_DOTNET_TRACER_CONFIG_FILE") ??
      Path.Combine(GetCurrentDirectory(), "datadog.json");
```

Reply:
👍 Then I would say the same mechanism be removed for `datadog.json` 😉

Reply:
Sounds reasonable. Should we refactor json configuration loading in different PR? - as this addresses completely different logic than current one. + the parsing issue https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/129#discussion_r622942562.

Reply:
Did you not mean https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/129#discussion_r622507858 ? 

For me, it can be a separate PR + GH issue (just to not forget about it)

Reply:
I'm currently documenting some configuration setup after a client issue. This reminds me that we actually cannot remove `datadog.json` autoload (although it should be renamed for OTel context). `OTEL_TRACE_CONFIG_FILE` default can be overloaded only via environment variable which transforms it to the global scope. Clients with multiple applications in the same environment do not have a way to configure app specific values after we remove autoload path. Autoload will ensure that `{App1Path}/datadog.json` and `{App2Path}/datadog.json` can be loaded.

Reply:
> can be overloaded only via environment variable which transforms it to the global scope

The env var setting could be scooped to a terminal scope or even single .NET process execution. Or am I missing something?

Reply:
We had the issue with IIS hosting multiple .NET Framework web apps, that could not be env scoped - although this issue was solved with web.config / app.config.
I'm not 100% sure, that everything could be solved with env scoping or app.config / web.config, there is loads of tech there that are not directly hostable. 

Maybe just add another var to enable config load? - so by default it will be off.

Reply:
> I'm not 100% sure, that everything could be solved with env scoping or app.config / web.config.
> [...] Maybe just add another var to enable config load? - so by default it will be off.

I would rather add a var to enable config load if we are certain that we cannot solve it any other way. _Less is more_ 😉

Reply:
I would agree if it's about adding functionality. But more concerned when we are reducing publicly available functionality. Anyway we can discuss it under a new issue when it's open and more detailed.

Reply:
I am ok with moving it to an issue and discussing it during SIG.